iib/workers/tasks/fbc_utils.py

```diff
@@ -87,3 +87,32 @@ def merge_catalogs_dirs(src_config: str, dest_config: str):
 
     log.info("Merging config folders: %s to %s", src_config, dest_config)
     shutil.copytree(src_config, dest_config, dirs_exist_ok=True)
+
+
+def extract_fbc_fragment(temp_dir: str, fbc_fragment: str) -> Tuple[str, str]:
+    """
+    Extract operator package from the fbc_fragment image.
+
+    :param str temp_dir: base temp directory for IIB request
+    :pararm str fbc_fragment: pull specification of fbc_fragment in the IIB request
+    :return fbc_fragment path, fbc_operator_package
+    :rtype: tuple
+    """
+    from iib.workers.tasks.build import _copy_files_from_image
+
+    log.info("Extracting the fbc_fragment's catalog from  %s", fbc_fragment)
+    # store the fbc_fragment at /tmp/iib-**/fbc-fragment
+    conf = get_worker_config()
+    fbc_fragment_path = os.path.join(temp_dir, conf['temp_fbc_fragment_path'])
+    # Copy fbc_fragment's catalog to /tmp/iib-**/fbc-fragment
+    _copy_files_from_image(fbc_fragment, conf['fbc_fragment_catalog_path'], fbc_fragment_path)
+
+    log.info("fbc_fragment extracted at %s", fbc_fragment_path)
+    operator_packages = os.listdir(fbc_fragment_path)
```

Comment:
Is our job to validate content? Test if fragment is valid (for example: test fragment does not contain only empty directory without json/yaml files) ...

Reply:
hmm, good point. I am not sure whether the built fbc_component will be directly used in IIB call or if there will be any validation tests on it before calling IIB. From what I know, the catalog.json/yaml file is actually created by using veneers (https://github.com/operator-framework/example-operator-index/blob/main/README.md?plain=1#L9), so I think it will be less likely that it will be empty, but it might be possible
How about we add this additional check in the next PR, by the time we will know how the Stonesoup pipeline is using the new endpoint?

Reply:
sure - it just popped in my mind so I raised this question

Reply:
"hmm maybe we do not need test because it will fail on cache generation?" possible